tests/framework/utils.py

```diff
@@ -206,3 +229,106 @@ def get_cpu_vendor():
     if 'AuthenticAMD' in brand_str:
         return CpuVendor.AMD
     return CpuVendor.INTEL
+
+
+def is_range(content):
+    """Return true if `content` is a range.
+
+    E.g ranges: 0-10.
+    """
+    match = re.search("([0-9][1-9]*)-([0-9][1-9]*)", content)
+    # Group is a singular value.
+    return match is not None
+
+
+def list_range(content):
+    """Return a range of integers based on the `content`.
+
+    The `content` respects the LIST FORMAT defined in the
+    cpuset documentation.
+    See: https://man7.org/linux/man-pages/man7/cpuset.7.html.
+    """
+    content = content.strip()
+    ends = content.split("-")
+    if len(ends) != 2:
+        return None
```

Comment:
Do we want to return `None` here? If `None` is appended to the array where this function is used, the behavior may trigger other errors, so I suggest either raising an exception or returning `[]`.